runtime/js/40_http.js

```diff
@@ -67,8 +67,9 @@
       const request = fromInnerRequest(innerRequest, "immutable");
 
       const respondWith = createRespondWith(responseSenderRid, this.#rid);
+      const upgradeWebSocket = createUpgradeWebSocket(requestBodyRid);
 
-      return { request, respondWith };
+      return { request, respondWith, upgradeWebSocket };
```

Comment:
instead of returning `upgradeWebsocket` here there should be a global `Deno.upgradeWebSocket` that takes a `Request` and returns both a `WebSocket(Stream)` and a `Response`. The connection will only be upgraded after the user passes this `Response` to `respondWith`.